073-kafka-crd-certalg-option.md

```diff
@@ -0,0 +1,51 @@
+# Enhance Kafka Spec with cert algorithm management
+
+Allow the end user to manage cert configs, used to generate servers and users certificates, via the `Kafka` CRD.
+
+## Current situation
+
+Properties are not yet supported in the Strimzi Kafka CRD.
+
+## Motivation
+
+Requested by the community here: https://github.com/strimzi/strimzi-kafka-operator/issues/9372
+
+## Proposal
+
+The proposal will introduce cert configs for the cluster and client CA, this is so that it is possible to use other algorithms than the currently hardcoded RSA. The proposal will add three new attributes to the `clusterCa` and `clientsCa` specs:
+
+* `keyAlgorithm`: The algorithm for generating the private key.
+* `keySize`: The size of the private key generated.
+* `signatureAlgorithm`: The hashing algorithm used for signing.
```

Comment:
We need more detail about these:

*Are they validated via the CRD Schema, or the Cluster Operator itself? Or do we just pass them to `openssl` and treat algorithms that are unknown to it as any other openssl failure? 
* Are we going to document some key algorithms which we guarantee will be available? How might we, in the future, go about deprecating and removing support for some combination of these parameters?
* Documenting allowed key sizes is tricky because they depend on the key algorithm. 
* Do the same settings apply for both CA and end-entity certificates?
* Are these settings only use for _new_ certificates, or does changing them imply that certificates need to be immediately regenerated?